### PR TITLE
split io and metrics generation

### PR DESCRIPTION
**Pull request by xudifsd**
split io operation with metrics generation

### CODE REVIEW

File: prometheus/exporter/job_exporter.py [hunk 1]
```diff
@@ -151,36 +144,45 @@ def wrapper(semaphore, queue):
...
+            gpuInfos = singleton.try_get()
+
+            gpuMetrics = gpu_exporter.convert_gpu_info_to_metrics(gpuInfos)
+            if gpuMetrics is not None:
+                utils.export_metrics_to_file(gpuMetricsPath, gpuMetrics)
+
             # join with docker stats metrics and docker inspect labels
-            gen_job_metrics(logDir, gpu_metrics)
+            jobMetrics = collect_job_metrics(gpuInfos)
+            utils.export_metrics_to_file(jobMetricsPath, jobMetrics)
```
Comment (YanjieGao) [inline prometheus/exporter/job_exporter.py]: not seen this utils code at master branch

Comment (xudifsd) [reply prometheus/exporter/job_exporter.py]: ah, I forget to add it.

**Comment by xudifsd** [comment]
sure

**Comment by xudifsd** [comment]
@YanjieGao fixed a bug found in int bed. should be ready for merge. Later we should add more test cases to increase case coverage to avoid having to test in int bed.

**Pull request merged**